### ISSUE TITLE
[Enhancement] Add Turbo Button Toggle

### ISSUE THREAD

**Issue by ichee** [libretro/RetroArch#11402]
Having a toggle for the designated turbo button would be useful for when you wish to keep a turbo button pressed long term, instead of having to find something to keep the button pressed, similar to how the 'fast forward' toggle currently works.

Also, at least with the Xbox One Wireless Controller, the controller will turn off after a period of inactivity, which also includes no further button presses, so keeping a button held without releasing or pressing another button will still result in the controller turning off after its timeout period.

**Comment by sonninnos** [comment]
With the "Single Button" mode that is already possible. One press and the selected button keeps getting smashed.

**Comment by ichee** [comment]
That's only for cores that support just one button. It isn't an option for most cores.

**Comment by sonninnos** [comment]
There is a select option for the default button though. Rarely one needs to keep autofiring multiple buttons at the same time, but even that is possible with the current "Single Button" mode.

**Comment by ichee** [comment]
Finally got it to work properly with beetle psx, but the configuration needed is somewhat elaborate.

The turbo button has to be configured first under `Input -> Port 1 Controls -> Turbo`, then the Turbo Default Button has to be set to the desired button, then the turbo mode can be changed to Single Button mode for the set turbo button to toggle correctly.

Rather than having it function that way, it would be better for the user set turbo button under `Input -> Port 1 Controls -> Turbo` to be both the toggle and the active turbo button.

**Comment by sonninnos** [comment]
Yes, the current is not optimal, but I have a few changes coming up regarding it:
https://github.com/libretro/RetroArch/pull/11550 & https://github.com/libretro/RetroArch/pull/11552

**Comment by LightAcolyte** [comment]
I have tried doing as ichee suggested in RetroArch on multiple emulators and cannot get this to work. For people with physical disabilities, it would be best this is extremely difficult to manage the button presses to set this up. The "Turbo Fire" should simply allow you to set the button so that when you press that button, it will treat it as it was being repeatedly pressed until you press that button again. With game overrides there is no need for anything else. If you want, allow a second option to allow toggling this functionality on and off with another button, but for people who really need additional buttons held please just have the basic option that every time you press the button once then after that it will treat it as being pressed for X frames out of a Y frame period infinitely and indefinitely. Once the same button is pressed again then it stops. Right now RetroArch makes the underlying emulator cores basically unplayable in RetroArch without this functionality for people who do not have the coordination or dexterity or manipulative appendages to do this. Thanks for your consideration!
